packages/worker/src/workflow/vm-shared.ts

```diff
@@ -338,7 +338,7 @@ export abstract class BaseVMWorkflow implements Workflow {
     await new Promise(setImmediate);
     if (this.unhandledRejection) {
       return {
-        runId: activation.runId,
+        runId: this.activator.info.runId,
```

Comment:
Curious why you changed this

Reply:
In `test-workflows.ts`, we set `this.activator.info.runId` to the title of the test, but `activation.runId` to `test-runId`.

Then, if the activation completes without issue, the completion response use `this.activator.info.runId`, but if it fails, it uses `activation.runId`.

Now, in test-workflows.ts' `activate()`, we do:

```
async function activate(t: ExecutionContext<Context>, activation: coresdk.workflow_activation.IWorkflowActivation) {
  const { workflow, runId } = t.context;
  const completion = await workflow.activate(activation);
  t.deepEqual(completion.runId, runId);   // <------ HERE
  return completion;
}
```

There, we compare the `completion.runId` to the expected `runId`, which corresponds to `this.activator.info.runId`. The sum of all of this is that if the activation fails with an exception, the `t.deepEquals()` will report an incorrect `runId` before the test code itself gets the chance to even inspect the content of the activation response itself. That makes it more difficult to locate the true reason of the failure.

This can be resolved inside `test-workflows.ts` instead, but it seems to me that `this.activator.info.runId` is really the correct thing to use, rather `activation.runId`.